libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1804,23 +1804,25 @@ pair<smtutil::Expression, smtutil::Expression> SMTEncoder::divModWithSlacks(
 {
 	IntegerType const* intType = &_type;
 	string suffix = "div_mod_" + to_string(m_context.newUniqueId());
-	smt::SymbolicIntVariable d(intType, intType, "d_" + suffix, m_context);
-	smt::SymbolicIntVariable r(intType, intType, "r_" + suffix, m_context);
+	smt::SymbolicIntVariable dSymb(intType, intType, "d_" + suffix, m_context);
+	smt::SymbolicIntVariable rSymb(intType, intType, "r_" + suffix, m_context);
+	auto d = dSymb.currentValue();
+	auto r = rSymb.currentValue();
 
 	// x / y = d and x % y = r iff d * y + r = x and
 	// either x >= 0 and 0 <= r < abs(y) (or just 0 <= r < y for unsigned)
 	// or     x < 0 and -abs(y) < r <= 0
-	m_context.addAssertion(((d.currentValue() * _right) + r.currentValue()) == _left);
+	m_context.addAssertion(((d * _right) + r) == _left);
```

Comment:
So as per my understanding `.currentValue()` generates a statement for the query (on first invocation), of which the order we want to be deterministic, but the ordering of these expression we don't care about.